src/core/server/saved_objects/service/saved_objects_client.ts

```diff
@@ -374,3 +375,15 @@ export class SavedObjectsClient {
     return await this._repository.update(type, id, attributes, options);
   }
 }
+
+export interface SavedObjectIndexPatternField {
+  name: string;
+  type: string;
+  aggregatable: boolean;
+  searchable: boolean;
+}
+
+export interface SavedObjectIndexPattern {
```

Comment:
The plural doesn't always make sense, but I've adopted the convention of always using SavedObject**s** to prefix type names.